src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -787,3 +797,17 @@ def parse_proxy_resource_id(rid):
         result.pop('children', None)
         return {key: value for key, value in result.items() if value is not None}
     return None
+
+
+def get_az_user_agent():
+    # Dynamically load the core version
+    from azure.cli.core import __version__ as core_version
+
+    agents = ["AZURECLI/{}".format(core_version)]
+
+    # msrest already has this
+    # https://github.com/Azure/msrest-for-python/blob/4cc8bc84e96036f03b34716466230fb257e27b36/msrest/pipeline/universal.py#L70
+    # if ENV_ADDITIONAL_USER_AGENT in os.environ:
+    #     agents.append(os.environ[ENV_ADDITIONAL_USER_AGENT])
+
+    return ' '.join(agents)
```

Comment:
May I know why we need the new function to handle agents? From my side, the previous UA definition should be fine, too.

Reply:
Because previously the functionality of concatenating UA is dispersed in 3 places:
- mgmt client
- data client
- az rest

It is difficult to maintain. I extrated the logic to one place so that changes done here automatically applies to all 3 places, given #12717 is doing more tweaks on the UA.